File: Fanscore_Backend/FanScore.Api/Models/Player.cs
using System;
using System.Collections.Generic;

namespace FanScore.Api.Models;

public partial class Player
{
    public int PlayerId { get; set; }

    public string Name { get; set; } = null!;

    public string Surname { get; set; } = null!;

    public int? TeamId { get; set; }

    public int? Age { get; set; }

    public string? Position { get; set; }

    public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();

    public virtual Team? Team { get; set; }
}

File: src/models/player.ts
export type PeriodKey = 'daily' | 'weekly' | 'monthly' | '3months' | '1year';
export type ViewMode = 'trends' | 'shuffle';

export interface Player {
  playerId: number;
  name: string;
  surname: string;
  teamId: number | null;
  teamName: string | null;
  age: number | null;
  position: string | null;
  ppUrl: string | null;
  averageRating: number | null;
  ratingCount: number;
  change: number | null;
}

export interface PlayerRating {
  ratingId: number;
  userId: number;
  ratingValue: number;
  comment: string | null;
  createdAt: string;
  periodType: string;
}

export interface PlayerDetail {
  playerId: number;
  name: string;
  surname: string;
  teamId: number | null;
  teamName: string | null;
  age: number | null;
  position: string | null;
  ppUrl: string | null;
  averageRating: number | null;
  ratingCount: number;
  ratings: PlayerRating[];
}

export interface HeatmapPlayer {
  id: number;
  name: string;
  team: string;
  position: string;
  averageRating: number;
  ratingCount: number;
  change: number;
  ppUrl?: string;
};
